applications/system/js_app/js_thread.c

```diff
@@ -197,13 +198,95 @@ static void js_require(struct mjs* mjs) {
 static void js_global_to_string(struct mjs* mjs) {
     int base = 10;
     if(mjs_nargs(mjs) > 1) base = mjs_get_int(mjs, mjs_arg(mjs, 1));
-    double num = mjs_get_int(mjs, mjs_arg(mjs, 0));
+    double num = mjs_get_double(mjs, mjs_arg(mjs, 0));
     char tmp_str[] = "-2147483648";
     itoa(num, tmp_str, base);
     mjs_val_t ret = mjs_mk_string(mjs, tmp_str, ~0, true);
     mjs_return(mjs, ret);
 }
 
+static void js_parse_int(struct mjs* mjs) {
+    mjs_val_t arg = mjs_arg(mjs, 0);
+    if(!mjs_is_string(arg)) {
```

Comment:
I recommend that you use the new JS ASS (Argument Schema for Scripts) macros, they will reduce these 11 lines of argument validation code down to just 1. Look at the new modules that I wrote, they heavily utilize ASS.

Also, use one of the new `strint_to_*` functions. They already implement this, you just need a JS wrapper around one.

Reply:
Saw this one coming lol, we've had these for a long time and never bothered to update it, I was debating fixing it before submitting the pr but decided to atleast submit it first to get some opinions

Reply:
i could not find a way to specify an optional argument and give it a default value using ASS, so for now i used an if check and used 2 ASS statements. looking at what it expands to, this looks wasteful (generates lots of duplicated code), but not sure what the best thing to do is